src/main/java/com/mraof/minestuck/item/StructureScannerItem.java

```diff
@@ -1,85 +1,183 @@
 
+import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.client.ClientProxy;
 import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.Gui;
+import net.minecraft.client.renderer.item.CompassItemPropertyFunction;
+import net.minecraft.client.renderer.item.ItemProperties;
 import net.minecraft.core.BlockPos;
+import net.minecraft.core.GlobalPos;
+import net.minecraft.nbt.NbtUtils;
 import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.sounds.SoundEvents;
 import net.minecraft.sounds.SoundSource;
 import net.minecraft.tags.TagKey;
 import net.minecraft.world.InteractionHand;
 import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.levelgen.structure.Structure;
 
+import javax.annotation.Nullable;
 import java.util.function.Supplier;
 
+/**
+ * Generic class for scanners which locates the nearest structure of a certain type in the player's current dimension.
+ * Takes an optional fuel item, which will be consumed every tick until the device deactivates.
+ * Toggles on or off with right-click, or when fuel runs out.
+ */
+
 public class StructureScannerItem extends Item
 {
 	private final TagKey<Structure> structure;
+	@Nullable
 	private final Supplier<Item> fuelItem;
 	
-	public StructureScannerItem(Properties properties, TagKey<Structure> structure, Supplier<Item> fuelItem)
+	public StructureScannerItem(Properties properties, TagKey<Structure> structure, @Nullable Supplier<Item> fuelItem)
 	{
 		super(properties);
 		this.structure = structure;
 		this.fuelItem = fuelItem;
 	}
 	
 	@Override
-	public InteractionResultHolder<ItemStack> use(Level levelIn, Player playerIn, InteractionHand handIn)
+	public InteractionResultHolder<ItemStack> use(Level pLevel, Player pPlayer, InteractionHand pUsedHand)
 	{
-		levelIn.playSound(playerIn, playerIn.getX(), playerIn.getY(), playerIn.getZ(), SoundEvents.UI_BUTTON_CLICK.value(), SoundSource.AMBIENT, 0.8F, 1.3F);
+		ItemStack stack = pPlayer.getItemInHand(pUsedHand);
 		
-		ItemStack item = playerIn.getItemInHand(handIn);
-		boolean foundItem = false;
+		if(pPlayer.isCreative() || stack.getDamageValue() != stack.getMaxDamage() && checkFuelNeeded(pPlayer, pLevel))
+		{
+			stack.getOrCreateTag().putBoolean("Powered", true);
+			pLevel.playSound(pPlayer, pPlayer.getX(), pPlayer.getY(), pPlayer.getZ(), SoundEvents.UI_BUTTON_CLICK.get(), SoundSource.AMBIENT, 0.8F, 1.3F);
+			
+			MutableComponent message = Component.translatable("message.temple_scanner.on");
+			pPlayer.sendSystemMessage(message.withStyle(ChatFormatting.DARK_GREEN));
```

Comment:
this should be server side only

Reply:
I put in a check along the lines of
`if (!sLevel.isClientSide)`
that seems to do the trick.